File: src/test/java/leets/weeth/domain/file/test/fixture/FileTestFixture.java
package leets.weeth.domain.file.test.fixture;

import leets.weeth.domain.board.domain.entity.Notice;
import leets.weeth.domain.file.domain.entity.File;

public class FileTestFixture {
    public static File createFile(Long id, String fileName, String fileUrl){
        return File.builder()
                .id(id)
                .fileName(fileName)
                .fileUrl(fileUrl)
                .build();
    }

    public static File createFile(Long id, String fileName, String fileUrl, Notice notice){
        return File.builder()
                .id(id)
                .fileName(fileName)
                .fileUrl(fileUrl)
                .notice(notice)
                .build();
    }
}

File: src/test/kotlin/com/weeth/domain/file/fixture/FileTestFixture.kt
package com.weeth.domain.file.fixture

import com.weeth.domain.file.domain.entity.File
import com.weeth.domain.file.domain.entity.FileOwnerType
import com.weeth.domain.file.domain.vo.FileContentType
import com.weeth.domain.file.domain.vo.StorageKey

object FileTestFixture {
    fun createFile(
        id: Long,
        fileName: String,
        storageKey: StorageKey = StorageKey("NOTICE/2026-02/test.png"),
        fileSize: Long = 1024,
        ownerType: FileOwnerType = FileOwnerType.NOTICE,
        ownerId: Long = 1L,
        contentType: FileContentType = FileContentType("image/png"),
    ): File =
        File(
            id = id,
            fileName = fileName,
            storageKey = storageKey,
            fileSize = fileSize,
            ownerType = ownerType,
            ownerId = ownerId,
            contentType = contentType,
        )
}
